cluster-operator/src/main/java/io/strimzi/operator/cluster/model/metrics/JmxPrometheusExporterModel.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.model.metrics;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.strimzi.api.kafka.model.common.HasConfigurableMetrics;
+import io.strimzi.api.kafka.model.common.metrics.JmxPrometheusExporterMetrics;
+import io.strimzi.operator.common.InvalidConfigurationException;
+import io.strimzi.operator.common.Reconciliation;
+import io.strimzi.operator.common.ReconciliationLogger;
+import io.strimzi.operator.common.model.InvalidResourceException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Model for the Prometheus JMX Exporter Java agent.
+ */
+public class JmxPrometheusExporterModel implements MetricsModel {
+    private static final ReconciliationLogger LOGGER = ReconciliationLogger.create(JmxPrometheusExporterModel.class.getName());
+
+    /**
+     * Key under which the metrics configuration is stored in the ConfigMap
+     */
+    public static final String CONFIG_MAP_KEY = "metrics-config.json";
+
+    private final boolean isEnabled;
+    private final String configMapName;
+    private final String configMapKey;
+
+    /**
+     * Constructs the Metrics Model for managing configurable metrics to Strimzi
+     *
+     * @param spec       Custom resource section configuring metrics
+     */
+    public JmxPrometheusExporterModel(HasConfigurableMetrics spec) {
+        if (spec.getMetricsConfig() != null) {
+            JmxPrometheusExporterMetrics config = (JmxPrometheusExporterMetrics) spec.getMetricsConfig();
+            validateJmxExporterMetricsConfig(config);
+            this.isEnabled = true;
+            this.configMapName = config.getValueFrom().getConfigMapKeyRef().getName();
+            this.configMapKey = config.getValueFrom().getConfigMapKeyRef().getKey();
+        } else {
+            this.isEnabled = false;
+            this.configMapName = null;
+            this.configMapKey = null;
```

Comment:
This is wrong. You have to throw an exception / raise an error and not just silently ignore it.

Reply:
Ok thanks for highlighting. Now refactored